src/cartservice/src/cartstore/RedisCartStore.cs

```diff
@@ -30,15 +41,72 @@ public RedisCartStore(IDistributedCache cache)
             _cache = cache;
         }
 
+        private static readonly HttpClient httpClient = new HttpClient();
+        
+        private async Task NotifySlackAsync(string productId)
+        {
+            var slackToken = Environment.GetEnvironmentVariable("SLACK_BOT_TOKEN");
+
+            if (string.IsNullOrWhiteSpace(slackToken))
+            {
+                Console.WriteLine("SLACK_BOT_TOKEN is not set in environment variables.");
+                return;
+            }
+
+            // Ensure we always have the correct auth header
+            httpClient.DefaultRequestHeaders.Remove("Authorization");
+            httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", slackToken);
+
+            var message = $"🚨 I have detected a 'CRITICAL' error within the cartservice when someone tried to add (productId: `{productId}`). Let me see what I can do. 👀";
+
+            // Manually construct JSON to avoid reflection-based serialization
+            var json = $"{{\"channel\":\"C08M5SMJ0KW\",\"text\":\"{message.Replace("\"", "\\\"")}\"}}";
+            var content = new StringContent(json, System.Text.Encoding.UTF8, "application/json");
+
+            var response = await httpClient.PostAsync("https://slack.com/api/chat.postMessage", content);
+            var responseBody = await response.Content.ReadAsStringAsync();
+
+            if (!response.IsSuccessStatusCode)
+            {
+                Console.WriteLine($"Slack API request failed: {response.StatusCode}");
+            }
+        }
+
+        private async Task CallExternalApiAsync()
+        {
+            var request = new HttpRequestMessage
+            {
+                Method = HttpMethod.Post,
+                RequestUri = new Uri("http://aa9781ba0ef2a41e68cdc7c3e7e6bb0e-1567177087.eu-west-2.elb.amazonaws.com/diagnose")
```

Comment:
Same here, do we want to parameterise it? Since it may change, it should be picked up from secrets probably.